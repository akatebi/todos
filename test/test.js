const {User, AddTodo, MarkAllTodos, ClearCompletedTodos} = require("./schema");

describe('Testing Todo GraphQL', () => {
    let userId;
    beforeAll(async() => {
        const email = "test@test.com";
        const resp = await User({email})
        // console.log("resp", JSON.stringify(resp, 0, 2));
        expect(resp).toMatchSnapshot();
        userId = resp.data.user.id;
    });
    test("AddTodo", async() => {
      const clientMutationId = "1";
      const text = "Get a customer";
      const resp = await AddTodo({text, userId, clientMutationId});
      // console.log("resp", JSON.stringify(resp, 0, 2));
      expect(resp).toMatchSnapshot();
    });
    test("MarkAllTodos", async() => {
      const clientMutationId = "2";
      const complete = true;
      const resp = await MarkAllTodos({complete, userId, clientMutationId});
      // console.log("resp", JSON.stringify(resp, 0, 2));
      expect(resp).toMatchSnapshot();
    });
    test("ClearCompletedTodos", async() => {
      const clientMutationId = "3";
      const resp = await ClearCompletedTodos({userId, clientMutationId});
      // console.log("resp", JSON.stringify(resp, 0, 2));
      expect(resp).toMatchSnapshot();
    });
  });