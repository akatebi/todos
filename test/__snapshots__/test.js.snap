// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "data": Object {
    "user": Object {
      "completedCount": 0,
      "email": "test@test.com",
      "id": "VXNlcjoy",
      "todos": Object {
        "edges": null,
        "pageInfo": Object {
          "endCursor": null,
          "hasNextPage": true,
        },
      },
      "totalCount": 0,
    },
  },
}
`;

exports[`Testing Todo GraphQL AddTodo 1`] = `
Object {
  "data": Object {
    "addTodo": Object {
      "clientMutationId": "1",
      "todoEdge": Object {
        "cursor": "NQ==",
        "node": Object {
          "complete": false,
          "id": "VG9kbzo1",
          "text": "Get a customer",
        },
      },
      "user": Object {
        "completedCount": 0,
        "email": "test@test.com",
        "todos": Object {
          "edges": Array [
            Object {
              "cursor": "NQ==",
              "node": Object {
                "complete": false,
                "id": "VG9kbzo1",
                "text": "Get a customer",
              },
            },
          ],
        },
        "totalCount": 1,
      },
    },
  },
}
`;

exports[`Testing Todo GraphQL ClearCompletedTodos 1`] = `
Object {
  "data": Object {
    "clearCompletedTodos": Object {
      "clientMutationId": "3",
      "deletedTodoIds": Array [
        "VG9kbzo1",
      ],
      "user": Object {
        "completedCount": 0,
        "id": "VXNlcjoy",
        "totalCount": 0,
      },
    },
  },
}
`;

exports[`Testing Todo GraphQL MarkAllTodos 1`] = `
Object {
  "data": Object {
    "markAllTodos": Object {
      "changedTodos": Array [
        Object {
          "complete": true,
          "id": "VG9kbzo1",
          "text": "Get a customer",
        },
      ],
      "clientMutationId": "2",
      "user": Object {
        "completedCount": 1,
        "id": "VXNlcjoy",
        "totalCount": 1,
      },
    },
  },
}
`;
